onnx/defs/shape_inference.h

```diff
@@ -340,9 +340,9 @@ inline const TensorShapeProto& getInputShape(const InferenceContext& ctx, size_t
   if (value_case != TypeProto::kTensorType && value_case != TypeProto::kSparseTensorType) {
     fail_type_inference("Input ", n, "expected to be a tensor or a sparse tensor type in ", ctx.getDisplayName(), ".");
   }
-  if (!hasShape(*input_type)) {
-    fail_shape_inference("Input ", n, " must have a non null shape in ", ctx.getDisplayName(), ".");
-  }
+
+  // do not check hasShape(*input_type) here, as scalar input is valid and it will return a default empty
```

Comment:
I'm wondering why this was not caught sooner. Do you have an example where it is needed? My guess is an error should be raised sooner. This change is not aligned with hasInputShape which requires hasShape to be true.